### DISCUSSION TITLE
Enum argument in command labelled as floating point

### DISCUSSION THREAD

**Discussion by pauladam316** [nasa/fprime#1874] [Q&A]
Hello! 

I'm having a strange issue with fprime-gds, I have a command defined that takes an enum as its argument. When I go to send the command through fprime-gds I can choose the enum through a menu as expected, however I get an error message stating that I must supply a floating point number

![image](https://user-images.githubusercontent.com/10899113/216168110-2e36f812-20b6-48cf-9756-54905006d421.png)

Is this a known bug? or has anyone seen this behavior before? I can't imagine what is causing it, its a very simple setup declaring a command that takes an enum as an argument.

Any help is always appreciated

**Comment by LeStarch** [comment]
This can happen when the first letter of the string is `F`.   i.e.  If your masked text above is F.....EnableAutomaticErrorLogging then this issue occurs. 

We've corrected the oversight in the latest GDS code.  So if you install one of our alpha pre-releases, it should fix the issue.

**Comment by pauladam316** [comment]
The first letter of the string is "e", would that also cause the issue?

**Comment by thomas-bc** [comment]
Does your fpp module name start with an F? That could trigger the issue too, I believe.

**Comment by pauladam316** [comment]
I think the issue was that my deployment started with F. I upgraded to an alpha build and it fixed that issue, but now I have a new one. My deployment logs are not appearing in fprime-gds

![image](https://user-images.githubusercontent.com/10899113/217098629-57963adc-e532-43b6-87e5-d09329e280cd.png)

is this a known issue?

**Reply by LeStarch** [reply]
The deployment logs only appear when you are running as `fprime-gds`  if you supply the `-n` argument, then they won't appear as you are independently running the deployment.  The logs are created using `deployment > deployment.log` thus they'll appear in the terminal if you are running it in the terminal directly.

Does this help?

**Reply by LeStarch** [reply]
I just checked and they aren't displaying for me either.  This is a bug.

**Reply by LeStarch** [reply]
Reported the bug here. https://github.com/nasa/fprime/issues/1885

**Comment by sriramyabhamidipati** [comment]
I see the same issue of "Supply floating point number" with my custom enum type in fprime-gds. I am running version 3.1.1. Looks like this issue is preventing the execution of the async command handler which considers this custom enum type as input. My namespace starts with an "F", which unfortunately cannot be changed I see that this issue is fixed in the latest version, but I cannot upgrade my version, unfortunately either. Is there a way to get around this problem?

**Reply by thomas-bc** [reply]
@sriramyabhamidipati you should be able to upgrade the version of the `fprime-gds` Python package without altering anything else. This bug has been fixed in recent versions of the GDS.
```
pip install -U fprime-gds
```
should do the trick.

**Reply by sriramyabhamidipati** [reply]
Yes, this works! thank you for the help.
